### PR TITLE
Improve performance of all screens using OptionsList

### PR DESCRIPTION
**Pull request by marcaaron**
### Details
We were rendering a very large initial set of data for anything using the `OptionsList`. I can't remember why it was set up like that, but removing it doesn't seem to have any negative effects so far.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/155388

### Tests
1. Click on the magnifying glass icon
2. Verify that the Search screen loads quickly

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

**Comment by marcaaron** [comment]
Tests well on web so far. Checking native platforms to be sure this doesn't explodey anything.

**Comment by marcaaron** [comment]
**Android performance before:**
`[Wed Mar 10 2021 11:15:20.526]  DEBUG    Timing:expensify.cash.search_render 3671`

**Android performance after:**
`[Wed Mar 10 2021 11:14:12.636]  DEBUG    Timing:expensify.cash.search_render 189`

File: src/components/OptionsList.js [hunk 1]
```diff
@@ -211,7 +211,6 @@ class OptionsList extends Component {
                     showsVerticalScrollIndicator={false}
                     sections={this.props.sections}
                     keyExtractor={this.extractKey}
-                    initialNumToRender={500}
```
Comment (Julesssss) [inline src/components/OptionsList.js:214]: > Note these items will never be unmounted as part of the windowed rendering in order to improve perceived performance of scroll-to-top actions.

!! Wow, nice spot.

**Review by Julesssss — DISMISSED** [review]
Insane improvement 😍, but I wonder if its a good idea to keep a small number here? From my brief testing the timing difference is trivial.

**Timing Averages**
- 'initialNumToRender:null' -- 30ms
- **'initialNumToRender: 15' -- 40ms**
- initialNumToRender: 500' -- 250ms

**Comment by Julesssss** [comment]
Also, it might be nice to add a new timing graph in the dashboard for this new event.

**Comment by marcaaron** [comment]
>wonder if its a good idea to keep a small number here? From my brief testing the timing difference is trivial

I'm not sure why we should? But I'm assuming there is some default initial number so I think we can just leave this for now 🤷‍♂️

**Comment by marcaaron** [comment]
Added it to the dash

**Comment by Julesssss** [comment]
> But I'm assuming there is some default initial number so I think we can just leave this for now 🤷‍♂️

I also assumed this, but after looking for some time (and testing some values), I don't think this is the case. 

But yeah, it's not necessary so lets leave it.

**Review by Julesssss — APPROVED** [review]

**Pull request merged**